### PR TITLE
Book Store - Setup Redux

### PR DESCRIPTION
**Pull request by aamirkhan2478**
Define action types for adding and removing a book.
- Set the initial state to be an empty array of books.
- Export Action Creators for your actions.

Write your reducer and export it as default.
- Define state changes for the actions that you created.
- In case of unknown action - return to the current state.

Define an action type for checking the status.
- Set the initial state to be an empty array of categories.
- Export Action Creators for your actions.

Write your reducer and export it as default.
- For the check status action return a string "Under construction".
- In case of unknown action - return to the current state.
- Configure the Redux Store (/src/Redux/store.js):

- Importing the necessary methods from Redux Toolkit.
- Combine both reducers into a root reducer by using the configureStore function.

### CODE REVIEW

File: src/Redux/Books/bookTypes.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const ADD_BOOK = 'ADD_BOOK';
+export const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (FrederickMih) [inline src/Redux/Books/bookTypes.js]: - [ ] Hi @aamirkhan2478, I think defining action type strings in a separate file is not a good coding practice. You should place these action types within your action creators and one or more of your reducers
> **Explanation**
  > - Defining the action types as constants and importing them wherever used reduces the risk of accidentally misspelling them in one of the sources in which they're used, which can cause a lot of wasted time debugging a simple bug/misspelling. Hence kindly move these action types into the `books.js` file where you are using them.

- [ ] Secondly based on the requirements of this project, your action types should follow the [duck pattern](https://github.com/erikras/ducks-modular-redux) methodology. This is very important especially if you have plans to scale your project in the future. Hence you'll need something like the line of code below:

```.ruby
export const ADD_BOOK = 'bookstore/books/ADD_BOOK';

```

File: src/Redux/Categories/categoriesTypes.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const SHOW_STATUS = 'SHOW_STATUS';
+
+export default SHOW_STATUS;
```
Comment (FrederickMih) [inline src/Redux/Categories/categoriesTypes.js]: - [ ] Kindly refer to the comments above and fix this. A lot of explanation of this practice has been outlined clearly above.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @aamirkhan2478 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

:arrow_forward:  Correct GitHub flow in this repo has been followed :heavy_check_mark: 
:arrow_forward: PR has a title and summary :heavy_check_mark: 

## Required Changes ♻️

- [ ]  Your README.md file is missing the last License section. This section is very vital because it renders your README file professional hence let's kindly include it.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @aamirkhan2478 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**